docs/specs/buffer.md

```diff
@@ -18,6 +18,20 @@ Vector buffers MUST be instrumented for optimal observability and monitoring. Th
 
 ### Events
 
+#### `BufferCreated`
+
+*All buffers* MUST emit a `BufferCreated` event immediately upon creation.
+
+* Properties
+  * `max_size` - the maximum number of events or byte size of the buffer
+  * `initial_events_size` - the number of events in the buffer at creation
+  * `initial_bytes_size` - the byte size of the buffer at creation
+  * `component_id` - the ID of the component associated with the buffer
```

Comment:
I realized, at the least, we should publish `component_scope` here too as I believe `id`s could collide if the scopes are different.

If we are able to include all of the normal component span information, that would still be preferable though.

Reply:
Gotcha. Would it be better to define a `component_metadata` property (e.g. some struct that would contain ID, scope, and potential others) here for flexibility?

Reply:
They end up flattened in the metric output, but for purposes of documentation that is fine if we define `component_metadata` somewhere in this doc and just refer to it for each metric.